### PR TITLE
more linters part5

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- enable gochecknoglobals
- enable gochecknoinits
- enable noctx
- enable godox
- enable ginkgolinter
- enable errorlint
- enable stylecheck
- enable gomnd
- enable gocritic
- enable thelper
- enable godot
- enable gocognit
- enable staticcheck
- enable goconst
- enable unparam
- enable forcetypeassert
- enable inamedparam
- enable unconvert
- enable varnamelen
- enable dupl
- enable nolintlint
- enable gosec
- enable tparallel

**Related issue(s)**

### CODE REVIEW

File: pkg/eventing/deployment.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func newDeployment(eventing *v1alpha1.Eventing, publisherConfig env.PublisherCon
 				Spec: kcorev1.PodSpec{
 					RestartPolicy:                 kcorev1.RestartPolicyAlways,
 					ServiceAccountName:            GetPublisherServiceAccountName(*eventing),
-					TerminationGracePeriodSeconds: &TerminationGracePeriodSeconds,
+					TerminationGracePeriodSeconds: &trmGrcPrd,
```
Comment (marcobebway) [inline pkg/eventing/deployment.go]: (nit) I see `utils.Int64Ptr` is used in the file.

File: hack/e2e/common/eventing/publisher.go [hunk 2]
```diff
@@ -64,9 +64,10 @@ func (p *Publisher) SendCloudEventWithRetries(event *cloudevents.Event, encoding
 }
 
 func (p *Publisher) SendLegacyEvent(source, eventType, payload string) error {
+	ctx := context.Background()
 	url := p.LegacyPublishEndpoint(source)
 
-	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBufferString(payload))
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBufferString(payload))
```
Comment (marcobebway) [inline hack/e2e/common/eventing/publisher.go]: 👍

File: internal/controller/eventing/subscription/jetstream/matchers_test.go [hunk 3]
```diff
@@ -35,16 +35,16 @@ func BeJetStreamSubscriptionWithSubject(source, subject string,
 		if err != nil {
 			return false, err
 		}
-		js := jetstream.JetStream{
+		jetstream := jetstream.JetStream{
```
Comment (marcobebway) [inline internal/controller/eventing/subscription/jetstream/matchers_test.go:38]: This shadows the package name `jetstream`.

File: internal/controller/operator/eventing/integrationtests/natsconnection/integration_test.go [hunk 4]
```diff
@@ -21,6 +21,8 @@ import (
 )
 
 // Test_NATSConnection tests the Eventing CR status when connecting to NATS.
+//
+//nolint:tparallel // lets not make this test parallel for now
```
Comment (marcobebway) [inline internal/controller/operator/eventing/integrationtests/natsconnection/integration_test.go:25]: I still see `t.Parallel()` in the test line `#91`. Was it failing the `tparallel` linter?

File: pkg/backend/jetstream/utils_internal_integration_test.go [hunk 5]
```diff
@@ -15,8 +15,8 @@ func TestJetStream_isJsSubAssociatedWithKymaSub(t *testing.T) {
 	// given
 	testEnvironment := setupTestEnvironment(t)
 	jsBackend := testEnvironment.jsBackend
-	defer testEnvironment.natsServer.Shutdown()
-	defer testEnvironment.jsClient.natsConn.Close()
+	t.Cleanup(testEnvironment.natsServer.Shutdown)
+	t.Cleanup(testEnvironment.jsClient.natsConn.Close)
```
Comment (marcobebway) [inline pkg/backend/jetstream/utils_internal_integration_test.go:19]: 👍

File: internal/controller/eventing/subscription/jetstream/matchers_test.go (hunk 3 above)
Comment (k15r) [reply internal/controller/eventing/subscription/jetstream/matchers_test.go:38]: fixed

**Review by marcobebway — APPROVED** [review]

**Pull request merged**